src/hdmf/backends/hdf5/h5tools.py

```diff
@@ -918,26 +918,30 @@ def __resolve_dtype_helper__(cls, dtype):
     def set_attributes(self, **kwargs):
         obj, attributes = getargs('obj', 'attributes', kwargs)
         for key, value in attributes.items():
-            if isinstance(value, (set, list, tuple)):
-                tmp = tuple(value)
-                if len(tmp) > 0:
-                    if isinstance(tmp[0], str):
-                        value = [np.unicode_(s) for s in tmp]
-                    elif isinstance(tmp[0], bytes):
-                        value = [np.string_(s) for s in tmp]
-                    elif isinstance(tmp[0], Container):  # a list of references
-                        self.__queue_ref(self._make_attr_ref_filler(obj, key, tmp))
-                    else:
-                        value = np.array(value)
-                self.logger.debug("Setting %s '%s' attribute '%s' to %s"
-                                  % (obj.__class__.__name__, obj.name, key, value.__class__.__name__))
-                obj.attrs[key] = value
-            elif isinstance(value, (Container, Builder, ReferenceBuilder)):           # a reference
-                self.__queue_ref(self._make_attr_ref_filler(obj, key, value))
-            else:
-                self.logger.debug("Setting %s '%s' attribute '%s' to %s"
-                                  % (obj.__class__.__name__, obj.name, key, value.__class__.__name__))
-                obj.attrs[key] = value                   # a regular scalar
+            try:
+                if isinstance(value, (set, list, tuple)):
+                    tmp = tuple(value)
+                    if len(tmp) > 0:
+                        if isinstance(tmp[0], str):
+                            value = [np.unicode_(s) for s in tmp]
+                        elif isinstance(tmp[0], bytes):
+                            value = [np.string_(s) for s in tmp]
+                        elif isinstance(tmp[0], Container):  # a list of references
+                            self.__queue_ref(self._make_attr_ref_filler(obj, key, tmp))
+                        else:
+                            value = np.array(value)
+                    self.logger.debug("Setting %s '%s' attribute '%s' to %s"
+                                      % (obj.__class__.__name__, obj.name, key, value.__class__.__name__))
+                    obj.attrs[key] = value
+                elif isinstance(value, (Container, Builder, ReferenceBuilder)):           # a reference
+                    self.__queue_ref(self._make_attr_ref_filler(obj, key, value))
+                else:
+                    self.logger.debug("Setting %s '%s' attribute '%s' to %s"
+                                      % (obj.__class__.__name__, obj.name, key, value.__class__.__name__))
+                    obj.attrs[key] = value                   # a regular scalar
+            except Exception as e:
```

Comment:
Why was this try/except added?

Reply:
To produce a better error message when it fails. It's not expected as a result of any common misuse, but just in case. It happened to me when I was working on this feature, and it was the only way I could figure out what was going on.